### PR TITLE
feat(embodied): add ResNet reward model

### PR DESCRIPTION
**Pull request by zhoutianxing7B**
### Description
Add ResNet-based reward model for embodied RL training. The reward model uses RGB images to predict task success/failure and provides learned rewards to guide policy learning.

Key features:
- `ResNetRewardModel`: ResNet18-based binary classifier for success/failure prediction
- `ImageRewardWorker`: Worker for reward inference during RL training
- `FSDPRewardWorker`: FSDP-based worker for reward model training
- Support for terminal reward (end of episode) and per-step reward modes
- Configurable reward mapping (asymmetric rewards to prevent policy collapse)

### Motivation and Context
Learned reward models can replace or augment sparse environment rewards, enabling:
- Training from visual observations without hand-crafted reward functions
- Transfer of reward models across similar tasks
- Reward shaping to improve sample efficiency

### How has this been tested?
- Tested reward model training on collected ManiSkill PickCube data
- Validated inference accuracy on held-out validation set
- Integrated with PPO training loop and verified reward computation
- Debug image saving for visual inspection of classification results

### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by QuanluZhang** [comment]
@guozhen1997 @andylin-hao as the primary reviewer

File: examples/embodiment/config/maniskill_ppo_mlp_collect.yaml [hunk 1]
```diff
@@ -0,0 +1,164 @@
+defaults:
```
Comment (XuS1994) [inline examples/embodiment/config/maniskill_ppo_mlp_collect.yaml]: collect yaml should not include actor env rollout config
